src/alt.js

```diff
@@ -219,18 +219,30 @@ const StoreMixin = {
 const setAppState = (instance, data, onStore) => {
   const obj = JSON.parse(data)
   Object.keys(obj).forEach((key) => {
-    assign(instance.stores[key][STATE_CONTAINER], obj[key])
-    onStore(instance.stores[key])
+    const store = instance.stores[key]
+    if (store[LIFECYCLE].deserialize) {
+      obj[key] = store[LIFECYCLE].deserialize(obj[key])
+      /* istanbul ignore next */
+      if(!obj[key]) {
```

Comment:
rather than throw an error how about we assign if it returns otherwise we don't assign? this allows you to do anything you want in deserialize without having the pressure of having to set the state yourself.